src/components/AnonymousChat/MessageChannelItem.tsx

```diff
@@ -49,6 +50,16 @@
       }
     : null;
 
+  const onFollowUser = () => {
+    followUserAction(item);
+  };
+
+  React.useEffect(() => {
+    if (!dbAnonUserInfo && item?.channelType === 'PM') {
+      getOtherProfileFollowStatus(item?.user?.id);
+    }
+  }, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onFollowUser` function and the `React.useEffect` hook seem to be implemented correctly. The `useEffect` hook will fetch the follow status of the other user when the component mounts if the channel type is 'PM' and there is no anonymous user info in the database. However, it would be better to add `item` and `dbAnonUserInfo` as dependencies to the `useEffect` hook to ensure it runs whenever these values change.

```diff
-  React.useEffect(() => {
-    if (!dbAnonUserInfo && item?.channelType === 'PM') {
-      getOtherProfileFollowStatus(item?.user?.id);
-    }
-  }, []);
+  React.useEffect(() => {
+    if (!dbAnonUserInfo && item?.channelType === 'PM') {
+      getOtherProfileFollowStatus(item?.user?.id);
+    }
+  }, [item, dbAnonUserInfo]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->